### PR TITLE
Add BD8LB600FS output diagnostics sensor

### PR DESCRIPTION
**Pull request by benediktibk**
Implement the output diagnostics of the low side switch BD8LB600FS as a sensor.

### CODE REVIEW

**Review by henrikbrixandersen — COMMENTED** [review]
I think this ought to be done using an MFD. The approach taken here is the same as before we had `drivers/mfd/`.

**Review by aasinclair — CHANGES_REQUESTED** [review]
Thanks for this, just one nit.

File: drivers/sensor/bd8lb600fs/bd8lb600fs_diagnostics.c [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+					      struct sensor_value *val)
+{
+	struct bd8lb600fs_diagnostics_data *data = dev->data;
+
+	switch (chan) {
+	case SENSOR_CHAN_BD8LB600FS_OPEN_LOAD:
+		val->val1 = data->old;
+		return 0;
+	case SENSOR_CHAN_BD8LB600FS_OVER_CURRENT_OR_THERMAL_SHUTDOWN:
+		val->val1 = data->ocp_or_tsd;
```
[the remark this answers is not in the document]
Comment (benediktibk) [reply drivers/sensor/bd8lb600fs/bd8lb600fs_diagnostics.c:39]: done

**Review by aasinclair — DISMISSED** [review]

**Comment by benediktibk** [comment]
@MaureenHelm Could you please take a look at this PR?

File: drivers/gpio/Kconfig.bd8lb600fs [hunk 2]
```diff
@@ -1,13 +1,13 @@
 # BD8LB600FS GPIO configuration options
 
-# Copyright (c) 2022 SILA Embedded Solutions GmbH
+# Copyright (c) 2024 SILA Embedded Solutions GmbH
```
Comment (MaureenHelm) [inline drivers/gpio/Kconfig.bd8lb600fs:3]: Add the new year without removing the previous one

File: drivers/gpio/gpio_bd8lb600fs.c [hunk 3]
```diff
@@ -1,111 +1,50 @@
 /*
- * Copyright (c) 2022 SILA Embedded Solutions GmbH
+ * Copyright (c) 2024 SILA Embedded Solutions GmbH
```
Comment (MaureenHelm) [inline drivers/gpio/gpio_bd8lb600fs.c:2]: Add the new year without removing the previous one

File: dts/bindings/gpio/rohm,bd8lb600fs-gpio.yaml [hunk 4]
```diff
@@ -1,5 +1,5 @@
 #
-# Copyright (c) 2022 SILA Embedded Solutions GmbH
+# Copyright (c) 2024 SILA Embedded Solutions GmbH
```
Comment (MaureenHelm) [inline dts/bindings/gpio/rohm,bd8lb600fs-gpio.yaml:2]: Add the new year without removing the previous one

File: drivers/sensor/bd8lb600fs/bd8lb600fs_diagnostics.c [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+		return -ENOTSUP;
+	}
+}
+
+static const struct sensor_driver_api bd8lb600fs_diagnostics_driver_api = {
+	.sample_fetch = bd8lb600fs_diagnostics_sample_fetch,
+	.channel_get = bd8lb600fs_diagnostics_channel_get,
+};
+
+int bd8lb600fs_diagnostics_init(const struct device *dev)
```
Comment (MaureenHelm) [inline drivers/sensor/bd8lb600fs/bd8lb600fs_diagnostics.c:53]: Make this function static

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Some minor nits, but otherwise LGTM

File: drivers/sensor/bd8lb600fs/bd8lb600fs_diagnostics.c (hunk 5 above)
Comment (benediktibk) [reply drivers/sensor/bd8lb600fs/bd8lb600fs_diagnostics.c]: done

File: dts/bindings/gpio/rohm,bd8lb600fs-gpio.yaml (hunk 4 above)
Comment (benediktibk) [reply dts/bindings/gpio/rohm,bd8lb600fs-gpio.yaml]: done

File: drivers/gpio/gpio_bd8lb600fs.c (hunk 3 above)
Comment (benediktibk) [reply drivers/gpio/gpio_bd8lb600fs.c]: done

**Review by MaureenHelm — APPROVED** [review]

**Review by teburd — APPROVED** [review]

**Comment by benediktibk** [comment]
CI failure is most likely unrelated to the changes in this PR

    INFO    - 767/767 qemu_x86_64               tests/modules/thrift/ThriftTest/thrift.ThriftTest.newlib.tlsTransport  FAILED unexpected eof (qemu 0.283s)
    INFO    - /__w/zephyr/zephyr/twister-out/qemu_x86_64/tests/modules/thrift/ThriftTest/thrift.ThriftTest.newlib.tlsTransport/handler.log
    ERROR   - SeaBIOS (version zephyr-v1.0.0-0-g31d4e0e-dirty-20200714_234759-fv-az50-zephyr)
    Booting from ROM..
    *** Booting Zephyr OS build v3.6.0-940-g71768defeda5 ***
    Running TESTSUITE thrift
    ===================================================================
    START - test_binary
    Thrift: Thu Jan  1 00:00:00 1970 TServerSocket::acceptImpl() ::accept() : Invalid argument
    Thrift: Thu Jan  1 00:00:00 1970 TSocket::open() connect() <Host: 192.0.2.1 Port: 4242>: Connection reset by peer
    abort()
    E: RAX: 0x0000000000000004 RBX: 0x00000000001dd910 RCX: 0x0000000000000001 RDX: 0x0000000000000001
    E: RSI: 0x0000000000193227 RDI: 0x00000000001d6720 RBP: 0x00000000001dd740 RSP: 0x00000000001dd740
    E:  R8: 0x0000000000000000  R9: 0x00000000001fb4d8 R10: 0x0000000000000574 R11: 0x00000000001f8930
    E: R12: 0x000000000017c373 R13: 0x00000000001ddcc0 R14: 0x0000000000000068 R15: 0x00000000001ddff8
    E: RSP: 0x00000000001dd740 RFLAGS: 0x0000000000000206 CS: 0x0018 CR3: 0x0000000000218000
    E: RIP: 0x000000000013fc1f
    E: >>> ZEPHYR FATAL ERROR 4: Kernel panic on CPU 0
    E: Current thread: 0x1b1640 (test_binary)
    E: Halting system

**Comment by benediktibk** [comment]
Ready to rumble (=merge)?

**Pull request merged**